lisa/sut_orchestrator/baremetal/bootconfig.py

```diff
@@ -0,0 +1,178 @@
+            expected_exit_code_failure_message=(
+                f"Failed to delete previous boot entry on {node_name}"
+            ),
+            expected_exit_code=0,
+            sudo=False,
+        )
+
+        self._log.debug(
+            "Adding boot entry for LISA at "
+            f"{pxe_boot_config_path}, "
+            f"pointing to {boot_entry}"
+        )
+
+        entry = self.pxe_runbook.kernel_append
+        append = f"\\n    append {entry}" if entry else ""
+        label = self._boot_label
+        self.pxe_server.execute(
+            cmd=f"sed -i '/^menu.*/a \\\\n{label}\\n    {boot_entry}{append}' "
+            f"{pxe_boot_config_path}",
+            expected_exit_code_failure_message=(
+                f"Failed to create new boot entry on {node_name}"
+            ),
+            expected_exit_code=0,
+            sudo=False,
+        )
+
+        self._log.debug(
+            "Added boot entry for LISA at "
+            f"{pxe_boot_config_path}, "
+            f"pointing to {boot_entry}"
+        )
+
+    def _connect_to_pxe_server(self) -> None:
+        assert self.pxe_runbook.connection, "connection is required for pxe_server"
+        self.pxe_runbook.connection.name = "pxe_server"
+        self.pxe_server = quick_connect(
+            self.pxe_runbook.connection, logger_name="pxe_server"
+        )
+
+    def _get_dhcp_info(self, node_name: str, host_ref: str) -> str:
+        output_dhcp_info = self.pxe_server.execute(
+            cmd="cat /etc/dhcp/dhcpd.conf",
+            expected_exit_code_failure_message=(
+                f"failed to obtain {node_name}'s configuration information\n"
+            ),
+            expected_exit_code=0,
+            sudo=False,
+        ).stdout
+        # Here is part of output_dhcp_info
+        # ...
+        # host dev-gp9 {
+        #    hardware ethernet 04:27:28:06:f7:88;
+        #    fixed-address 192.168.3.123;
+        #    option host-name "blade3";
+        # }
+        #
+        # host dev-gp10 {
+        #    hardware ethernet 04:27:28:15:3f:f0;
+        #    fixed-address 192.168.3.117;
+        #    option host-name "blade4";
+        # }
+        # ...
+        # if node 10 is used to be bootup from pxe_server,
+        # configuration file for node 10 on pxe_server needs
+        # to be modified. The configuration file's name for
+        # node 10 is actually based on its physical address:
+        # 04:27:28:15:3f:f0 as you can see from the above,
+        # and it is called 01-04-27-28-15-3f-f0 under bootup
+        # directory. Below is to find the node's physical
+        # address string 04:27:28:15:3f:f0 from output_dhcp_info
+        # and then obtain the configuration file name for
+        # later bootup menu change.
+        ethern_part = r"\s\{[\r\n]?\s+hardware ethernet\s+"
+        host_part = re.escape(host_ref) + ethern_part
+        node_str = host_part + r"([0-9a-f:]+)"
```

Comment:
One line is clearer, and you can capture both host and address, then find it in Python code. And you can test pattern, examples online like https://regex101.com/.
```
re.compile(r"^host\s+(?P<device>.+)\s+\{\n\s+hardware ethernet\s(?P<address>[0-9a-f:]+);$")
```

Reply:
The change is done, thanks